Description: Change Bonobo activation server name to avoid conflict with indicator-session
Bug: https://bugs.edge.launchpad.net/ubuntu/+source/gdm/+bug/410498

Index: gdm-2.30.4/gui/user-switch-applet/applet.c
===================================================================
--- gdm-2.30.4.orig/gui/user-switch-applet/applet.c	2010-06-26 00:09:31.000000000 +0200
+++ gdm-2.30.4/gui/user-switch-applet/applet.c	2010-06-30 12:38:51.000000000 +0200
@@ -101,7 +101,7 @@
                                 const char    *iid,
                                 gpointer       data);
 
-PANEL_APPLET_BONOBO_FACTORY ("OAFIID:GNOME_FastUserSwitchApplet_Factory",
+PANEL_APPLET_BONOBO_FACTORY ("OAFIID:GNOME_FastUserSwitchAppletGdm_Factory",
                              PANEL_TYPE_APPLET,
                              "gdm-user-switch-applet", "0",
                              (PanelAppletFactoryCallback)applet_factory,
@@ -1611,7 +1611,7 @@
 {
         gboolean ret;
         ret = FALSE;
-        if (strcmp (iid, "OAFIID:GNOME_FastUserSwitchApplet") == 0) {
+        if (strcmp (iid, "OAFIID:GNOME_FastUserSwitchAppletGdm") == 0) {
                 ret = fill_applet (applet);
         }
         return ret;
Index: gdm-2.30.4/gui/user-switch-applet/GNOME_FastUserSwitchApplet.server.in.in
===================================================================
--- gdm-2.30.4.orig/gui/user-switch-applet/GNOME_FastUserSwitchApplet.server.in.in	2010-06-16 18:32:38.000000000 +0200
+++ gdm-2.30.4/gui/user-switch-applet/GNOME_FastUserSwitchApplet.server.in.in	2010-06-30 12:38:51.000000000 +0200
@@ -1,5 +1,5 @@
 <oaf_info>
-  <oaf_server iid="OAFIID:GNOME_FastUserSwitchApplet_Factory" type="exe"
+  <oaf_server iid="OAFIID:GNOME_FastUserSwitchAppletGdm_Factory" type="exe"
               location="@LIBEXECDIR@/gdm-user-switch-applet">
 
     <oaf_attribute name="repo_ids" type="stringv">
@@ -18,8 +18,8 @@
     </oaf_attribute>
   </oaf_server>
 
-  <oaf_server iid="OAFIID:GNOME_FastUserSwitchApplet" type="factory"
-              location="OAFIID:GNOME_FastUserSwitchApplet_Factory">
+  <oaf_server iid="OAFIID:GNOME_FastUserSwitchAppletGdm" type="factory"
+              location="OAFIID:GNOME_FastUserSwitchAppletGdm_Factory">
 
     <oaf_attribute name="repo_ids" type="stringv">
       <item value="IDL:GNOME/Vertigo/PanelAppletShell:1.0"/>
